clang/lib/CIR/CodeGen/CIRGenTypes.cpp

```diff
@@ -303,14 +305,23 @@ const CIRGenFunctionInfo &CIRGenTypes::arrangeCIRFunctionInfo() {
   assert(!cir::MissingFeatures::opCallCallConv());
 
   // Construction the function info. We co-allocate the ArgInfos.
-  fi = CIRGenFunctionInfo::create();
+  fi = CIRGenFunctionInfo::create(returnType);
```

Comment:
It looks like `functionInfos` is the one that ends up keeping long-term track of this for memoization, but that doesn't seem to do the destruction either.